client/components/email-verification/verify-email-dialog.jsx

```diff
@@ -34,7 +37,7 @@ class VerifyEmailDialog extends React.Component {
 
 		this.setState( { pendingRequest: true } );
 
-		this.props.user.sendVerificationEmail( function( error, response ) {
+		user.sendVerificationEmail( function( error, response ) {
```

Comment:
The underlying implementation in the end calls`this.wpcom.req.post( { path: '/me/send-verification-email' }, callback );`. I would avoid setting up a user object here just for that. It's confusing in the context that this component requires a user prop.

The refactor looks large enough here to add a redux action for email verification, and update the existing usage before updating the gravatar behavior here.